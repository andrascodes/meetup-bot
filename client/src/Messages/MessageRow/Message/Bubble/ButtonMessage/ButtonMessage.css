.ButtonMessage {
  background-color: #fff;
  color: #000;
  border-radius: 1em;
  border: 0.2em solid rgba(0, 0, 0, .10);
}

.ButtonMessageText {
  background-color: #f1f0f0;
  padding: 0.4em;
  border-radius: 1em 1em 0em 0em;
}

.ButtonMessageButton {
  padding: 0.3em 0em;
  color: #0084ff;
  font-weight: 500;
}

.ButtonDataContainer {
  border-bottom: 0.1em solid rgba(0, 0, 0, .10);
}