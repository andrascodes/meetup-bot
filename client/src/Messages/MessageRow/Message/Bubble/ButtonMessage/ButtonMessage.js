import React from 'react'

import './ButtonMessage.css'

export const ButtonMessage = props => (
  <div className="ButtonMessage">
    <div className="ButtonMessageText ButtonDataContainer">
      {props.attachment.payload.text}
    </div>
    {props.attachment.payload.buttons.map((btn, ind) => (
      <div className="ButtonMessageButton ButtonDataContainer text-center" key={ind}>
        {btn.title}
      </div>
    ))}
  </div>
)