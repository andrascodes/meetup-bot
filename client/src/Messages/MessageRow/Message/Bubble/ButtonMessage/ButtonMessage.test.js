import React from 'react'
import renderer from 'react-test-renderer'

import { ButtonMessage } from './ButtonMessage'

describe('ButtonMessage', () => {

  test('renders correctly', () => {
    const tree = renderer.create(
      <ButtonMessage 
        attachment={{
          payload: {
            text: 'Message Text',
            buttons: [
              {
                title: 'Btn title1',
              },
              {
                title: 'Btn title2',
              },
            ]
          }
        }}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

})