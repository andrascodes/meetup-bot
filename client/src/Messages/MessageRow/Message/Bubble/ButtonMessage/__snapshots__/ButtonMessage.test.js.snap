exports[`ButtonMessage renders correctly 1`] = `
<div
  className="ButtonMessage">
  <div
    className="ButtonMessageText ButtonDataContainer">
    Message Text
  </div>
  <div
    className="ButtonMessageButton ButtonDataContainer text-center">
    Btn title1
  </div>
  <div
    className="ButtonMessageButton ButtonDataContainer text-center">
    Btn title2
  </div>
</div>
`;
